docs/hub/datasets-webdataset.md

```diff
@@ -0,0 +1,20 @@
+# WebDataset
+
+[WebDataset](https://github.com/webdataset/webdataset) is a library to write I/O pipelines for large datasets. 
+Since it supports streaming data using HTTP, you can use the Hugging Face data files URLs to stream a dataset in WebDataset format:
+
+First login using 
+
+```
+huggingface-cli login
+```
+
+And then you can stream Hugging Face datasets in WebDataset:
+
+```python
+>>> import webdataset as wds
+>>> from huggingface_hub import HfFolder
+
+>>> hf_token = HfFolder().get_token()
+>>> dataset = wds.WebDataset(f"pipe:curl -s -L https://huggingface.co/datasets/username/my_wds_dataset/resolve/main/train-000000.tar -H 'Authorization:Bearer {hf_token}'")
```

Comment:
I think it would be helpful to have the same structure and use cases for all the libraries, ie: here explicitly show how to read a dataset, and how to write it.

By the way, maybe we should refer to a real dataset for the "read" part, so that it always work? (not sure) ie: the structure would be:
- read a public dataset
- login
- create and write a user dataset
- read the user dataset

Reply:
IMO no need to have the same structure, as we add more and more libs we will want to show the focus on their main uses cases which might not always fit a common structure